app-api/src/main/java/org/openmrs/eip/app/management/entity/SenderSyncMessage.java

```diff
@@ -0,0 +1,142 @@
+package org.openmrs.eip.app.management.entity;
+
+import java.util.Date;
+
+import javax.persistence.Access;
+import javax.persistence.AccessType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+
+@Entity
+@Table(name = "sender_sync_message")
+public class SenderSyncMessage extends AbstractEntity {
+	
+	private static final long serialVersionUID = 1L;
+	
+	// TODO add ACKNOWLEDGED status, which will be set after successful processing of the receiver response
+	public enum SenderSyncMessageStatus {
+		NEW,
+		SENT,
+		ERROR
+	}
+	
+	@NotNull
+	@Column(name = "table_name", length = 100, nullable = false, updatable = false)
+	private String tableName;
+	
+	@NotNull
+	@Column(length = 255, nullable = false, updatable = false)
+	private String identifier;
+	
+	@NotNull
+	@Column(length = 1, nullable = false, updatable = false)
+	private String operation;
+	
+	@NotNull
+	@Column(name = "message_uuid", length = 38, nullable = false, unique = true, updatable = false)
+	private String messageUuid;
+	
+	@Column(name = "request_uuid", length = 38, updatable = false)
+	private String requestUuid;
+	
+	@NotNull
+	@Column(nullable = false, updatable = false)
+	private boolean snapshot;
+	
+	@NotNull
+	@Enumerated(EnumType.STRING)
+	@Column(nullable = false, length = 20)
+	@Access(AccessType.FIELD)
+	private SenderSyncMessageStatus status = SenderSyncMessageStatus.NEW;
+	
+	@Column(name = "date_updated")
+	@Access(AccessType.FIELD)
+	private Date dateUpdated;
+	
+	@Column(name = "error_message", length = 200)
+	@Access(AccessType.FIELD)
+	private String errorMessage;
```

Comment:
Why do you have the error_message column? It is not supposed to be there

Reply:
Before this implementation there was a TODO in the sender/dbsync-route.xml. See the version of the commit 8667f83535be2c1dad29587224f4c4c5a147f988:

```
            <otherwise>
                <!-- TODO Log to a special failures log file or DB -->
                <log message="No entity found in the database matching identifier: ${exchangeProperty.event.identifier}" loggingLevel="ERROR" />
            </otherwise>
```

So, in order to implement this TODO, we've added this column that is filled in the sender-activemq-publisher route, when no entity is found:

```
                        <otherwise>
                            <setProperty name="errorMessage">
                                <constant>No entity found in the database matching identifier</constant>
                            </setProperty>
                            
                            <log message="${exchangeProperty.errorMessage}: ${exchangeProperty.senderSyncMessage.identifier}" loggingLevel="ERROR" />
                            <script>
                                <spel>
                                    #{getProperty('senderSyncMessage').markAsError(getProperty('errorMessage'))}
                                </spel>
                            </script>
                        </otherwise>
```

So, if this column is not supposed to be there, please tell what we should do for the SenderSyncMessages that during the process in the sender-activemq-publisher, does not have the correspondent entity in the database for the identifier. Then I can implement accordingly.

Reply:
Resolved in [this discussion](https://github.com/FriendsInGlobalHealth/openmrs-eip/pull/11#discussion_r905321859)